client_auth.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+type AuthGetter interface {
+	Request(identity, deviceID string) (interface{}, error)
+}
+
+// source of authorization data
+type AuthDataProvider interface {
+	// returns device public key
+	Pubkey() (string, error)
+	// returns tenant token
+	Token() (string, error)
+	// returns identity data
+	IdentityData() (interface{}, error)
+}
+
+type AuthClient struct {
+	client *http.Client
```

Comment:
Actually both the update client and auth client have different interfaces, not sure if wrapping that up into a single client will help. However, what might help is sharing a single `http.Client`. This should allow us to benefit from pipelining as the client is supposed to keep the connections open.